code/game/objects/effects/overlays.dm

```diff
@@ -35,8 +35,7 @@
 	if(randomdir)
 		setDir(pick(cardinal))
 	flick("[icon_state]", src) //Because we might be pulling it from a pool, flick whatever icon it uses so it starts at the start of the icon's animation.
-	spawn(duration)
-		qdel(src)
+	QDEL_IN(src, duration)
```

Comment:
This is going to be called every time we use an overlay, we could save a lot through using the timer.